devscripts/make_changelog.py

```diff
@@ -0,0 +1,452 @@
+from __future__ import annotations
+
+import enum
+import itertools
+import json
+import logging
+import re
+import subprocess
+import sys
+from collections import defaultdict
+from dataclasses import dataclass
+from functools import cache
+from pathlib import Path
+
+BASE_URL = 'https://github.com'
+LOCATION_PATH = Path(__file__).parent
+
+logger = logging.getLogger(__name__)
+
+
+class CommitGroup(enum.Enum):
+    PRIORITY = 'Important'
+    CORE = 'Core'
+    EXTRACTOR = 'Extractor'
+    DOWNLOADER = 'Downloader'
+    POSTPROCESSOR = 'Postprocessor'
+    MISC = 'Misc.'
+
+    @classmethod
+    @cache
```

Comment:
Normally, I'd do `utils.classproperty`, but I don't think it's worth importing yt_dlp for just that